### PR TITLE
Set task name for async methods

### PR DESCRIPTION
**Pull request by hsorbo**

### CODE REVIEW

File: codegen/valagasyncmodule.vala [hunk 1]
```diff
@@ -261,6 +261,21 @@ public class Vala.GAsyncModule : GtkModule {
 
 		ccode.add_assignment (new CCodeMemberAccess.pointer (data_var, "_async_result"), create_result);
 
+		var full_method_name = new StringBuilder ();
+		full_method_name.append (m.name.has_prefix (".") ? m.name.substring (1) : m.name);
```
Comment (oleavr) [inline codegen/valagasyncmodule.vala:265]: Interesting -- a method name prefixed with `.`, when is this the case?

**Review by oleavr — CHANGES_REQUESTED** [review]
w00t! :metal:

File: codegen/valagasyncmodule.vala (hunk 1 above)
Comment (hsorbo) [reply codegen/valagasyncmodule.vala:265]: constructors. Initially i just checked if they where constructors and returned "new", but then i found out constructors can have name
from test finish-name.vala
```public class Foo {
	[CCode (finish_name = "foo_new_end")]
	public async Foo.async () {
	}

	[CCode (finish_name = "foo_bar_end")]
	public async string bar () {
		return "bar";
	}
	[CCode (finish_name = "foo_baz_end", finish_vfunc_name = "baz_end")]
	public virtual async string baz () {
		return "baz";
	}
}```

File: codegen/valagasyncmodule.vala [hunk 2]
```diff
@@ -261,6 +261,21 @@ public class Vala.GAsyncModule : GtkModule {
...
+		Vala.Symbol? symbol = m.parent_symbol;
+		while (symbol != null) {
+			if (symbol.name != null) {
```
[the remark this answers is not in the document]
Comment (hsorbo) [reply codegen/valagasyncmodule.vala]: fixed

File: codegen/valagasyncmodule.vala [hunk 3]
```diff
@@ -261,6 +261,21 @@ public class Vala.GAsyncModule : GtkModule {
...
+				full_method_name.prepend ("%s.".printf (symbol.name));
+			}
+			symbol = symbol.parent_symbol;
+		}
+
+		var set_name = new CCodeFunctionCall (new CCodeIdentifier ("g_task_set_name"));
+		set_name.add_argument (new CCodeMemberAccess.pointer (data_var, "_async_result"));
+		set_name.add_argument (new CCodeConstant("\"%s\"".printf (full_method_name.str)));
```
[the remark this answers is not in the document]
Comment (hsorbo) [reply codegen/valagasyncmodule.vala:275]: seems there is some history for doing it this way: https://github.com/frida/vala/blob/2c7a02d29332dedfdf803ce21b0540bc8461a035/codegen/valaccodebasemodule.vala#L5432

[the remark this answers is not in the document]
Comment (hsorbo) [reply codegen/valagasyncmodule.vala]: Fixed the missing space in the original

**Pull request merged**